### PR TITLE
Add partner events box

### PR DESCRIPTION
**Pull request by mattandrews**
Looks like this:

screen shot 2015-02-19 at 10 25 39

cc @jamesoram / @jennysivapalan

### CODE REVIEW

File: frontend/app/model/EventPortfolio.scala [hunk 1]
```diff
@@ -2,7 +2,15 @@ package model
...
 
-case class EventPortfolio(orderedEvents: Seq[RichEvent], normal: Seq[RichEvent], pastEvents: Option[Seq[RichEvent]]) {
+// used for arbitrary groupings of events with custom titles
+case class EventGroup(sequenceTitle: String, events: Seq[RichEvent])
+
+case class EventPortfolio(
+    orderedEvents: Seq[RichEvent],
+    normal: Seq[RichEvent],
+    pastEvents: Option[Seq[RichEvent]],
+    otherEvents: Option[EventGroup] = None
```
Comment (jennysivapalan) [inline frontend/app/model/EventPortfolio.scala]: I'd get rid of the `=None` as it is specified in both services.

File: frontend/app/services/EventbriteService.scala [hunk 2]
```diff
@@ -138,7 +146,7 @@ object GuardianLiveEventService extends EventbriteService {
 
   override def getFeaturedEvents: Seq[RichEvent] = EventbriteServiceHelpers.getFeaturedEvents(eventsOrderingTask.get(), events)
   override def getTaggedEvents(tag: String): Seq[RichEvent] = events.filter(_.name.text.toLowerCase.contains(tag))
-
+  override def getPartnerEvents = Some(EventGroup("Partner Events", events.filter(_.providerOpt.isDefined)))
```
Comment (jennysivapalan) [inline frontend/app/services/EventbriteService.scala]: To be consistent with the methods above we should add the type to the method?

File: frontend/app/services/EventbriteService.scala [hunk 3]
```diff
@@ -174,6 +182,7 @@ object MasterclassEventService extends EventbriteService {
 
   override def getFeaturedEvents: Seq[RichEvent] = Nil
   override def getTaggedEvents(tag: String): Seq[RichEvent] = events.filter(_.tags.contains(tag.toLowerCase))
+  override def getPartnerEvents = None
```
Comment (jennysivapalan) [inline frontend/app/services/EventbriteService.scala]: Ditto comment about  type to the method.

File: frontend/test/services/EventbriteServiceTest.scala [hunk 4]
```diff
@@ -41,6 +41,7 @@ class EventbriteServiceTest extends PlaySpecification {
     override def eventsArchive: Seq[RichEvent] = Nil
     override def getFeaturedEvents: Seq[RichEvent] = Nil
     override def getTaggedEvents(tag: String): Seq[RichEvent] = Nil
+    override def getPartnerEvents = None
```
Comment (jennysivapalan) [inline frontend/test/services/EventbriteServiceTest.scala]: Ditto

File: frontend/app/model/EventPortfolio.scala (hunk 1 above)
Comment (mattandrews) [inline frontend/app/model/EventPortfolio.scala]: When I do that I get complaints about not enough arguments in `Event.scala`...

**Comment by mattandrews** [comment]
@jennysivapalan have changed those things!

**Comment by jennysivapalan** [comment]
:+1: :european_castle:

**Comment by mattandrews** [comment]
:briefcase:

**Comment by davidrapson** [comment]
:+1:

**Comment by jamesoram** [comment]
:+1:

**Pull request merged**